nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,272 @@
+        description = ''
+          interval in seconds at which throughput should be reported [TS_PROM_TPUT_REPORT]
+        '';
+      };
+
+      web-listen-address = mkOption {
+        type = types.str;
+        default = ":9201";
+        description = ''
+          Address to listen on for web endpoints. [TS_PROM_WEB_LISTEN_ADDRESS] (default ":9201")
+        '';
+      };
+
+      web-telemetry-path = mkOption {
+        type = types.str;
+        default = "/metrics";
+        description = ''
+          Address to listen on for web endpoints. [TS_PROM_WEB_TELEMETRY_PATH] (default "/metrics")
+        '';
+      };
+
+      extraOptions = mkOption {
+        default = [];
+        type = types.listOf types.str;
+        example = literalExample ''
+          [ "-async-acks"
+          ]
+        '';
+        description = ''
+          Extra command line arguments to pass to timescale-prometheus.
+        '';
+      };
+
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = (cfg.db-host != "" && cfg.db-host != "localhost") -> (cfg.db-passwordFile != null);
+        message = "When services.timescale-prometheus.db-host is changed from default, services.timescale-prometheus.db-passwordFile must be set!";
+      }
+    ];
+
+    meta.maintainers = with lib.maintainers; [ _0x4A6F ];
+
+    systemd.services.timescale-prometheus = {
+      description = "An open-source analytical platform for Prometheus metrics";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
+      after = [ "postgresql.service" ];
+
+      environment = {
+        TS_PROM_ASYNC_ACKS = "${toString cfg.async-acks}";
+        TS_PROM_DB_CONNECT_RETRIES = "${toString cfg.db-connect-retries}";
+        TS_PROM_DB_HOST = "${toString cfg.db-host}";
+        TS_PROM_DB_NAME = "${toString cfg.db-name}";
+        TS_PROM_DB_PORT = "${toString cfg.db-port}";
+        TS_PROM_DB_SSL_MODE = "${toString cfg.db-ssl-mode}";
+        TS_PROM_DB_USER = "${toString cfg.db-user}";
+        TS_PROM_LABELS_CACHE_SIZE = "${toString cfg.labels-cache-size}";
+        TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_ID = "${toString cfg.leader-election-pg-advisory-lock-id}";
+        TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_PROMETHEUS_TIMEOUT = "${toString cfg.leader-election-pg-advisory-lock-prometheus-timeout}";
+        TS_PROM_LEADER_ELECTION_REST = "${toString cfg.leader-election-rest}";
+        TS_PROM_LOG_LEVEL = "${toString cfg.log-level}";
+        TS_PROM_METRICS_CACHE_SIZE = "${toString cfg.metrics-cache-size}";
+        TS_PROM_MIGRATE = "${toString cfg.migrate}";
+        TS_PROM_SCHEDULED_ELECTION_INTERVAL = "${toString cfg.scheduled-election-interval}";
+        TS_PROM_TPUT_REPORT = "${toString cfg.tput-report}";
+        TS_PROM_WEB_LISTEN_ADDRESS = "${toString cfg.web-listen-address}";
+        TS_PROM_WEB_TELEMETRY_PATH = "${toString cfg.web-telemetry-path}";
+      };
+
+      script = "
```

Comment:
Why not just use `serviceConfig.ExecStart`?

Reply:
`script` usage:
```
exec ${cfg.package}/bin/example \
  ${optionalString (cfg.passwordFile != null) "--password $(<${cfg.passwordFile})"}
```
- [x] no leak to store
- [x] no leak to process list (ps fauxwww | grep password)

---

`serviceConfig.ExecStart` usage:
```
${cfg.package}/bin/example \
  ${optionalString (cfg.passwordFile != null) "--password $(head -n 1 ${cfg.passwordFile})"}
```
- [x] no leak to store
- [?] no leak to process list (ps fauxwww | grep password)